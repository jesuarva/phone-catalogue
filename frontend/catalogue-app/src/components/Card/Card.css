.custom-card {
  min-height: 30vh;
}

.custom-card a h1 {
  text-decoration: none !important;
}