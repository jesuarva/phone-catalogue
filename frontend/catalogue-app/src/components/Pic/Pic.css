.pic img {
  min-height: 50vh;
}