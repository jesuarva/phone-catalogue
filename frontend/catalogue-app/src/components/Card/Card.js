import React from 'react';
import { Link } from 'react-router-dom';
import { Jumbotron } from 'reactstrap';
import Pic from '../Pic/Pic';
import './Card.css';

const Card = ({ phone }) => {
  const { image, Model } = phone;
  const urlModel = Model.replace(/\s/g, '-');
  return (
    <div className="col-12 col-md-4 custom-card">
      <Link to={`/catalogue/${urlModel}`}>
        <Jumbotron className="custom-card">
          <Pic img={image} altAttribute={Model} />
          <h1 className="display-6">{Model}</h1>
        </Jumbotron>
      </Link>
    </div>
  );
};

export default Card;
