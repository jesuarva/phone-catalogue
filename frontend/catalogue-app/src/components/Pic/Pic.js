import PropTypes from 'prop-types';
import React from 'react';
import './Pic.css';

const Pic = ({ img, altAttribute }) => {
  return (
    <div className="pic">
      <figure className="figure">
        <img
          src={img}
          className="figure-img img-fluid rounded"
          alt={altAttribute}
        />
      </figure>
    </div>
  );
};

Pic.propTypes = {
  img: PropTypes.string.isRequired,
  altAttribute: PropTypes.string.isRequired,
};

export default Pic;
