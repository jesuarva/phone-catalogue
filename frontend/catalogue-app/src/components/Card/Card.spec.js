import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Jumbotron } from 'reactstrap';
import Pic from '../Pic/Pic';
import TestWrapper from '../TestWrapper/TestWrapper';
import Card from './Card';

configure({ adapter: new Adapter() });

describe('Card component', () => {
  function setUp() {
    const props = {
      phone: {
        image:
          'https://store.storeimages.cdn-apple.com/4981/as-images.apple.com/is/image/AppleInc/aos/published/images/i/ph/iphone8/select/iphone8-select-2017?wid=212&hei=292&fmt=png-alpha&.v=1503618523038',
        Model: 'Apple iPhone 8 Plus',
      },
    };

    const testingComponent = (
      /**
       * <Router/> is neccesary in order to avoid the following error
       * "Invariant Violation: You should not use <Link> outside a <Router>"
       *
       */
      <Router>
        <Card {...props} />
      </Router>
    );

    return {
      props,
      testingComponent,
    };
  }

  it('should renders correctly', () => {
    const { testingComponent } = setUp();
    const tree = renderer.create(testingComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Renders children component', () => {
    const { testingComponent } = setUp();

    const mounted = mount(<TestWrapper>{() => testingComponent}</TestWrapper>);

    it('should renders a Link component', () => {
      const link = mounted.find(Link);
      expect(link).toHaveLength(1);
    });

    it('should renders a Jumbotron component', () => {
      const jumbo = mounted.find(Jumbotron);
      expect(jumbo).toHaveLength(1);
    });

    it('should renders a Pic component', () => {
      const pic = mounted.find(Pic);
      expect(pic).toHaveLength(1);
    });
  });
});
