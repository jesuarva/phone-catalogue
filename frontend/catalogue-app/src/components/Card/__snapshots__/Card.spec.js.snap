// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should renders correctly 1`] = `
<div
  className="col-12 col-md-4 custom-card"
>
  <a
    href="/catalogue/Apple-iPhone-8-Plus"
    onClick={[Function]}
  >
    <div
      className="custom-card jumbotron"
    >
      <div
        className="pic"
      >
        <figure
          className="figure"
        >
          <img
            alt="Apple iPhone 8 Plus"
            className="figure-img img-fluid rounded"
            src="https://store.storeimages.cdn-apple.com/4981/as-images.apple.com/is/image/AppleInc/aos/published/images/i/ph/iphone8/select/iphone8-select-2017?wid=212&hei=292&fmt=png-alpha&.v=1503618523038"
          />
        </figure>
      </div>
      <h1
        className="display-6"
      >
        Apple iPhone 8 Plus
      </h1>
    </div>
  </a>
</div>
`;
